packages/docgen/test/get-type-annotation.js

```diff
@@ -415,6 +415,68 @@ describe( 'Type annotations', () => {
 			).toBe( 'string[]' );
 		} );
 
+		it( 'should lean on `getDependants` if `selector` in `createSelector` is missing args', () => {
+			const { tokens } = engine(
+				'test.ts',
+				`/**
+				 * Returns a reference that changes only when the dependencies do.
+				 *
+				 * @param state - stores all the things
+				 */
+				 export const getCountReference = createSelector( () => [], ( state: State ) => [ state.items.length ] );
+				 `
+			);
+
+			expect(
+				getTypeAnnotation(
+					{ tag: 'param', name: 'state' },
+					tokens[ 0 ],
+					0
+				)
+			).toBe( 'State' );
+		} );
+
+		it( 'should ignore `getDependants` if `selector` in `createSelector` has more arguments', () => {
```

Comment:
I don't really understand what this test is testing :sweat_smile: Maybe the assertion is unclear to me. Is it proving that the argument passed in the `getDependents` is ignored when the selector itself has types?

For testing that... wouldn't we want both the state arguments to have the same name at least, given that docgen traverses based on the token names?

Reply:
I'm probably not fully understanding your question so I'll share some context in hope that it helps us get to the same page.

This is a tricky function with our docs because `selector` and `getDependants` (the arguments to `createSelector`) must accept the same parameters but will likely have different return types. That means we don't want separate `@param` tags for the two functions because their arguments have to be the same.

```ts
/**
 * @param selectorState
 * @param selectorType
 * @param selectorId
 * @param getDependantsState
 * @param getDependantsType
 * @param getDependantsId
 */
```

This is flat-out _wrong_ anyway because the JSDoc comments aren't describing `selector` or `getDependants` at all; they're describing the return value of `createSelector`.

What I'm trying to demonstrate in these tests and capture in the PR update is that we care about that full set of parameters while acknowledging that they don't have to be fully handled in each of the two functions. The base assumption, which led to this PR, is that `selector` will specify all the parameters. In practice, theory aside, this is not true, noted in the description.

Therefore we want to ask what the arguments are in `getDependants` if the `selector` is incomplete in specifying the arguments that will need to be supplied by calling code, but we want to make sure that we don't default to that because I suspect that in most cases `selector` will specify more arguments than `getDependants`.

I'm trying to balance the line between adding complexity and handling a rare but legitimate case, so I agree we want to ensure the state name is the same, but I don't want to add a bunch of code that tries to resolve a discrepancy without any real way to reliably automatically resolve it, but make a practical call in certain cases that may b knowably _less wrong_.

This could happen, but I suspect it's not a big problem in practice.
```ts
createSelector( state => state.posts, (postState, type id) => [type, id, postState.posts] )
```

At least I have only seen cases where both were called `state` or one was omitted.